skimage/data/_fetchers.py

```diff
@@ -209,6 +221,17 @@ def _fetch(data_filename):
     try:
         resolved_path = image_fetcher.fetch(data_filename)
     except ConnectionError as err:
+        # Check if pytest is currently running.
+        # Packagers might use pytest to run the tests suite, but may not
+        # want to run it online.
+        # As such, we will avoid failing the test, and silently skipping it.
+        if 'PYTEST_CURRENT_TEST' in os.environ:
+            # https://docs.pytest.org/en/latest/example/simple.html#pytest-current-test-environment-variable
+            import pytest
+            # Pytest skip raises an exception that allows the
+            # tests to be skipped
+            pytest.skip(f'Unable to download {data_filename}',
+                        allow_module_level=True)
```

Comment:
since this looks duplicate with the above, perhaps refactor into a small, private helper function?